### DISCUSSION TITLE
merge trouble

### DISCUSSION THREAD

**Discussion by HIwamaLyrist** [py-pdf/pypdf#1687] [Show and tell]
I want to make 2-up pages.
When I use ver.2, I can use "mergeRotatedScaledTranslatePage". Then I get what I want.

But, the function is deleted on ver.3.0.
So, I changed source.
Then I cannot get what I want.
I think it is merge trouble.
2nd page is cut by right blank page.

Could you please give me any advice?

## Environment
```bash
$ python -m platform
macOS-11.2.3-x86_64-i386-64bit

$ python -c "import pypdf;print(pypdf.__version__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'pypdf'

$ pip list
Package    Version
---------- -------
pdf2image  1.16.2
Pillow     9.4.0
pip        23.0.1
PyPDF2     2.12.1
setuptools 65.5.0

```

## Code + PDF
for PyPDF2 ver3
```python
import PyPDF2
from PyPDF2 import Transformation
import os
import sys

a3_width = 1190.5511811024
a3_height = 841.8897637795

a4_width = a3_width / 2
a4_height = a3_height

file_path = sys.argv[1]
pdf_file = open(file_path, 'rb')
pdf_source = PyPDF2.PdfReader(pdf_file)
max_page = len(pdf_source.pages)
pdf_write = PyPDF2.PdfWriter()
pdf_write_a4 = PyPDF2.PdfWriter()

dir_name = os.path.dirname(file_path)
file_path = os.path.splitext(os.path.basename(file_path))[0]

base_page = PyPDF2.PageObject.create_blank_page(width = a3_width, height=a3_height)
base_page.merge_page(pdf_source.pages[0])

#base_page = pdf_source.pages[0]


right_page = pdf_source.pages[1]
op = Transformation().translate(tx=a3_width/4*(1))
right_page.add_transformation(op)
base_page.merge_page(right_page)

pdf_write.add_page(base_page)



pdf_output_file = open(dir_name + '/' + file_path + '_2up_ver3.pdf', 'wb')
pdf_write.write(pdf_output_file)
pdf_output_file.close()


pdf_file.close()
```

for PyPDF2 ver2
```python
import PyPDF2
from PyPDF2 import Transformation
import os
import sys

a3_width = 1190.5511811024
a3_height = 841.8897637795

a4_width = a3_width / 2
a4_height = a3_height

file_path = sys.argv[1]
pdf_file = open(file_path, 'rb')
pdf_source = PyPDF2.PdfReader(pdf_file)
max_page = len(pdf_source.pages)
pdf_write = PyPDF2.PdfWriter()
pdf_write_a4 = PyPDF2.PdfWriter()

dir_name = os.path.dirname(file_path)
file_path = os.path.splitext(os.path.basename(file_path))[0]

base_page = PyPDF2.PageObject.create_blank_page(width = a3_width, height=a3_height)
base_page.merge_page(pdf_source.pages[0])
right_page = pdf_source.pages[1]
base_page.mergeRotatedScaledTranslatedPage(right_page,0, 1, a3_width /2, 0, expand=False)
pdf_write.add_page(base_page)



pdf_output_file = open(dir_name + '/' + file_path + '_2up_ver2.pdf', 'wb')
pdf_write.write(pdf_output_file)
pdf_output_file.close()


pdf_file.close()
```

[test_2up_ver2.pdf](https://github.com/py-pdf/pypdf/files/10877381/test_2up_ver2.pdf)
[test_2up_ver3.pdf](https://github.com/py-pdf/pypdf/files/10877382/test_2up_ver3.pdf)
[test.pdf](https://github.com/py-pdf/pypdf/files/10877383/test.pdf)

**Comment by pubpub-zz** [comment]
@HIwamaLyrist 
this is the code I propose you:
```
import pypdf
r = pypdf.PdfReader("test.pdf")
w = pypdf.PdfWriter()
w.append(r,[0])
p=w.pages[0]
p.merge_translated_page(r.pages[1],p.mediabox.width,0,True,True)
w.write("tst_2up.pdf")
```
[tst_2up.pdf](https://github.com/py-pdf/pypdf/files/10885616/tt4.pdf)

**Reply by higamma** [reply]
I forgot I changed the account for private.
I'm HIwamaLyrist.
may I ask some questions?
I strange 2 points.

1.
append(**r, [0]**) 
I can’t find how-to use append like this.
Could you tell me the name of this technique?
2.
**p.merge_translated_page(r.pages[1],p.mediabox.width,0,True,True)**
**w.write("tst_2up.pdf")**
Why object w is changed?

**Comment by HIwamaLyrist** [comment]
@pubpub-zz 
thx, your advice.
unfortunately, it's not work.

the following error is occurred.
AttributeError: 'PdfWriter' object has no attribute 'append'

Later, I found another page.
I refer it, and I got what I want.
[https://qiita.com/yusuke_s_yusuke/items/c0ab5b24bff9bbf86116]

**Comment by pubpub-zz** [comment]
You are using an obsolète version :you have to move to pypdf no more pyPDF2

**Comment by HIwamaLyrist** [comment]
@pubpub-zz 
thx.

i use pip to install.
Should I change the install tool?

**Comment by pubpub-zz** [comment]
No but the installation command line is
`pip install pypdf`
You'd better remove pyPDF2 before
`pip uninstall pyPDF2`

**Comment by HIwamaLyrist** [comment]
@pubpub-zz 
It's works!! thanks a lot.

**Comment by pubpub-zz** [comment]
for knowledge sharing, I will convert it into a discussion
